source.js

```diff
@@ -37,6 +37,15 @@ var SVGtoPDF = function(doc, svg, x, y, options) {
         if (Object.keys(doc.page.xobjects).length) { group.resources.data.XObject = doc.page.xobjects; }
         if (Object.keys(doc.page.ext_gstates).length) { group.resources.data.ExtGState = doc.page.ext_gstates; }
         if (Object.keys(doc.page.patterns).length) { group.resources.data.Pattern = doc.page.patterns; }
+        group.resources.data.ColorSpace = (function() {
+            var output = {};
+            var i = 1;
+            for (var item in doc.spotColors) {

```

Comment:
All this should be done in PDFKit.
Here it would be only:
`if (Object.keys(doc.page.color_spaces).length) { group.resources.data.ColorSpace = doc.page.color_spaces; }`

I commented some ideas how to do it on your pdfkit fork.